src/Orleans/Core/Grain.cs

```diff
@@ -272,9 +275,25 @@ internal string CaptureRuntimeEnvironment()
     /// Base class for a Grain with declared persistent state.
     /// </summary>
     /// <typeparam name="TGrainState">The interface of the persistent state object</typeparam>
-    public class Grain<TGrainState> : Grain
+    public class Grain<TGrainState> : Grain, IStorage
         where TGrainState : class, IGrainState
     {
+        private IStorage storage;
+
+        private IStorage Storage
+        {
+            get
+            {
+                if (storage == null) // Of storage has been injected via DI, we'll use it. Otherwise, we initialize here.
```

Comment:
I understand that this makes it easier but it looks more like a work around to avoid plumbing the storage. Instantiating the `IStorage` implementation should not be done in the `Grain`. I would recommend plumbing the `IStorage` the same way we did with other IGrainRuntime components as the current approach can have side effects.